capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -0,0 +1,129 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+
+import com.google.sps.data.Club;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that allows users to post and get comments. */
+@WebServlet("/api/clubs")
+public class ClubServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String stringID = request.getParameter("id");
+    Query query = new Query("Club");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long id = Long.parseLong(stringID);
+
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Entity> entities = results.asList(FetchOptions.Builder.withLimit(5));
```

Comment:
Looks like you're returning the first five clubs here. Is there a plan for the ClubServlet's `doGet` to do something different in the future, maybe return the clubs the user is in for example?

Reply:
We do eventually plan to grab clubs based on either search parameters (search by name, book reading) or by membership (see all clubs you're a member of, owner of, or were invited to). As we were discussing the move to Firebase, I decided to leave this as is for now since it may all have to be changed anyways. I just chose 5 as some arbitrary number to stand in in the meantime.